src/sentry/monitors/tasks.py

```diff
@@ -81,18 +81,24 @@ def check_monitors(current_datetime=None):
     )
     metrics.gauge("sentry.monitors.tasks.check_monitors.timeout_count", qs.count())
     # check for any monitors which are still running and have exceeded their maximum runtime
-    for checkin in RangeQuerySetWrapper(qs):
-        timeout = timedelta(minutes=(checkin.monitor.config or {}).get("max_runtime") or TIMEOUT)
-        if checkin.date_updated > current_datetime - timeout:
+    for (
+        checkin_id,
+        checkin_monitor,
+        checkin_monitor_environment,
```

Comment:
Ahh I missed that we're using the whole monitor env object. This should still work ok, just keep an eye on the stats when you deploy to make sure we're not ooming or anything.